Silicon/TigerlakePkg/Library/Stage2SocInitLib/Stage2SocInitLib.c

```diff
@@ -0,0 +1,98 @@
+/** @file
+  The platform hook library.
+
+  Copyright (c) 2013 - 2015, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <PiPei.h>
+#include <Base.h>
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/PciLib.h>
+#include <Library/PcdLib.h>
+#include <GlobalNvsAreaDef.h>
+#include <Library/IoLib.h>
+#include <GlobalNvsAreaDef.h>
+#include <Register/SaRegs.h>
+#include <Library/VTdLib.h>
+#include <PlatformData.h>
+
+/**
+  Get size of global nvs area.
+
+  @param[in] none
+
+  @retval    UINT32     Size of global nvs area.
+
+**/
+UINT32
+EFIAPI
+GetAcpiGnvsSize (
+  VOID
+)
+{
+  return sizeof (GLOBAL_NVS_AREA);
+}
+
+
+/**
+ Update GPIO address and length to global NVS data.
+
+ @param [in] GnvsIn Pointer to Global NVS data.
+ **/
+VOID
+EFIAPI
+SocUpdateAcpiGnvs (
+  IN VOID  *GnvsIn
+)
+{
+  ///
+  /// Vtd Initialization
+  ///
+  UINTN                           McD0BaseAddress;
+  UINTN                           McD2BaseAddress;
+  GLOBAL_NVS_AREA                 *Gnvs;
+  UINT8                           Index;
+  PLATFORM_DATA                   *PlatformData;
+
+  if (FeaturePcdGet (PcdVtdEnabled)) {
+    PlatformData = (PLATFORM_DATA *)GetPlatformDataPtr ();
+    if (PlatformData != NULL) {
+      if (PlatformData->PlatformFeatures.VtdEnable == 0) {
+        return;
+      }
+    }
+
+    Gnvs = (GLOBAL_NVS_AREA *)GnvsIn;
+    DEBUG ((DEBUG_INFO, "Initialize VT-d\n"));
+    McD0BaseAddress  = PCI_LIB_ADDRESS (SA_MC_BUS, 0, 0, 0);
+    McD2BaseAddress  = PCI_LIB_ADDRESS (SA_IGD_BUS, SA_IGD_DEV, SA_IGD_FUN_0, 0);
+
+   // TODO:  Need to decide if we have to add VtdDisable Silicon UPD here.
+    Gnvs->SaNvs.VtdDisable = !((FeaturePcdGet (PcdVtdEnabled)));

```

Comment:
```Gnvs->SaNvs.VtdDisable = FALSE``` since this code is already inside ```if (FeaturePcdGet (PcdVtdEnabled))```.

Reply:
Noted, I will change it and also remove the //TODO comment